x/auth/ante_test.go

```diff
@@ -627,3 +627,24 @@ func TestConsumeSignatureVerificationGas(t *testing.T) {
 		})
 	}
 }
+
+func TestAdjustFeesByGas(t *testing.T) {
+	type args struct {
+		fee sdk.Coins
+		gas int64
+	}
+	tests := []struct {
+		name string
+		args args
+		want sdk.Coins
+	}{
+		{"nil coins", args{sdk.Coins{}, 10000}, sdk.Coins{}},
```

Comment:
Should we have test cases for positive coins and one's that equal just `0`?

Reply:
We can't have that, see `types/coin.go`:

```
// Plus combines two sets of coins
// CONTRACT: Plus will never return Coins where one Coin has a 0 amount.
func (coins Coins) Plus(coinsB Coins) Coins {
[snip]
```

Reply:
I meant when `requiredFees - userFees = 0`, but I guess you already have that?